File: backend/src/main/java/com/echonovel/service/AuthorService.java
package com.echonovel.service;

import com.echonovel.dto.request.AuthorRequest;
import com.echonovel.dto.response.AuthorResponse;

import java.util.List;

/**
 * Service interface for author operations.
 */
public interface AuthorService {

    /**
     * Get all authors.
     */
    List<AuthorResponse> getAllAuthors();

    /**
     * Create a new author (Admin).
     */
    AuthorResponse createAuthor(AuthorRequest request);

    /**
     * Update an author (Admin).
     */
    AuthorResponse updateAuthor(Long id, AuthorRequest request);

    /**
     * Delete an author (Admin).
     */
    void deleteAuthor(Long id);
}

File: frontend/src/services/authorService.ts
import api from './api';
import type { ApiResponse, AuthorResponse, AuthorRequest } from '../types';

export const authorService = {
  getAuthors: async (): Promise<ApiResponse<AuthorResponse[]>> => {
    const response = await api.get<ApiResponse<AuthorResponse[]>>('/authors');
    return response.data;
  },
  
  createAuthor: async (data: AuthorRequest): Promise<ApiResponse<AuthorResponse>> => {
    const response = await api.post<ApiResponse<AuthorResponse>>('/admin/authors', data);
    return response.data;
  },

  updateAuthor: async (id: number, data: AuthorRequest): Promise<ApiResponse<AuthorResponse>> => {
    const response = await api.put<ApiResponse<AuthorResponse>>(`/admin/authors/${id}`, data);
    return response.data;
  },

  deleteAuthor: async (id: number): Promise<ApiResponse<void>> => {
    const response = await api.delete<ApiResponse<void>>(`/admin/authors/${id}`);
    return response.data;
  },
};
